api/src/services/applications/application_validation.py

```diff
@@ -171,7 +171,9 @@ def validate_application_form(
     process_rule_schema_for_context(context)
     form_validation_errors.extend(context.validation_issues)
 
-    # TODO pre/post-populate should update the value here
+    # Apply post-population changes to the application form during submit
+    if action == ApplicationAction.SUBMIT and context.config.do_post_population:
+        application_form.application_response = context.json_data
```

Comment:
It'll seem like potentially too much, but we actually want to just always do:
```suggestion
    application_form.application_response = context.json_data
```
The configuration itself already determines whether we actually run pre/post population rules. With pre-population disabled (until the next ticket) we _shouldn't_ ever be change anything outside of submit. When we do add pre-population, we'll have a lot of places where we do this.